File: Data Structures & Algorithms/maximum-subarray/submission-0.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        best = curr = nums[0]
        for x in nums[1:]:
            # either extend the current subarray or start new at x
            curr = max(x, curr + x)
            best = max(best, curr)
        return best

File: Data Structures & Algorithms/house-robber/submission-0.go
func rob(nums []int) int {
    n := len(nums)
    maxRob := make([]int, n+1)
    maxRob[1] = nums[0]
    for i := 2; i <= n; i++ {
        maxRob[i] = max(maxRob[i-1], maxRob[i-2]+nums[i-1])
    }
    return maxRob[n]
}
